### ISSUE TITLE
list index out of range

### ISSUE THREAD

**Issue by ralphbean** [fedora-infra/fedimg#21]
Saw this come across the sysadmin-datanommer emails:

```
Message
-------
[2015-03-14 12:57:20][    fedmsg   ERROR]
{'body': {u'username': u'apache', u'certificate': u'LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUVTakNDQTdPZ0F3SUJBZ0lDQWRzd0RRWUpL\nb1pJaHZjTkFRRUZCUUF3Z2FBeEN6QUpCZ05WQkFZVEFsVlQKTVFzd0NRWURWUVFJRXdKT1F6RVFN\nQTRHQTFVRUJ4TUhVbUZzWldsbmFERVhNQlVHQTFVRUNoTU9SbVZrYjNKaApJRkJ5YjJwbFkzUXhE\nekFOQmdOVkJBc1RCbVpsWkcxelp6RVBNQTBHQTFVRUF4TUdabVZrYlhObk1ROHdEUVlEClZRUXBF\nd1ptWldSdGMyY3hKakFrQmdrcWhraUc5dzBCQ1FFV0YyRmtiV2x1UUdabFpHOXlZWEJ5YjJwbFkz\nUXUKYjNKbk1CNFhEVEUxTURFd05USXhNemd5TVZvWERUSTFNREV3TWpJeE16Z3lNVm93Z2RneEN6\nQUpCZ05WQkFZVApBbFZUTVFzd0NRWURWUVFJRXdKT1F6RVFNQTRHQTFVRUJ4TUhVbUZzWldsbmFE\nRVhNQlVHQTFVRUNoTU9SbVZrCmIzSmhJRkJ5YjJwbFkzUXhEekFOQmdOVkJBc1RCbVpsWkcxelp6\nRXJNQ2tHQTFVRUF4TWlhMjlxYVMxcmIycHAKTURFdWNHaDRNaTVtWldSdmNtRndjbTlxWldOMExt\nOXlaekVyTUNrR0ExVUVLUk1pYTI5cWFTMXJiMnBwTURFdQpjR2g0TWk1bVpXUnZjbUZ3Y205cVpX\nTjBMbTl5WnpFbU1DUUdDU3FHU0liM0RRRUpBUllYWVdSdGFXNUFabVZrCmIzSmhjSEp2YW1WamRD\nNXZjbWN3Z1o4d0RRWUpLb1pJaHZjTkFRRUJCUUFEZ1kwQU1JR0pBb0dCQU9PaSt4VEsKSStQdEU2\nM!
 ll0ZkkzM3
 g4bW5UUE1ZU0hDZHB3NkhCNjZlbU1qeHRyMU95a082alY4NERQS1JDdzVFWCswUEV2\nbgozNFo2MCtKOXkxMS9JZzRuYTBhNlA5aVI5TlNXb1o3cENwQ3NMRUZIeFJUM0tqVlg4V3JkeURX\ndVRaSXNBN0c3CkVSL2wyRzlKaDBKSmVCNDBBcm9LMTJEeGllMjZ1dFQ5SHJpN0FnTUJBQUdqZ2dG\nWE1JSUJVekFKQmdOVkhSTUUKQWpBQU1DMEdDV0NHU0FHRytFSUJEUVFnRmg1RllYTjVMVkpUUVNC\nSFpXNWxjbUYwWldRZ1EyVnlkR2xtYVdOaApkR1V3SFFZRFZSME9CQllFRkxNVGpXekFPOURmYU83\nT1h0K2VkdFVMd2I3Vk1JSFZCZ05WSFNNRWdjMHdnY3FBCkZHdEFXdmtTQ0lsWjUxbmxCZlVDSFFw\nT2Z4UUFvWUdtcElHak1JR2dNUXN3Q1FZRFZRUUdFd0pWVXpFTE1Ba0cKQTFVRUNCTUNUa014RURB\nT0JnTlZCQWNUQjFKaGJHVnBaMmd4RnpBVkJnTlZCQW9URGtabFpHOXlZU0JRY205cQpaV04wTVE4\nd0RRWURWUVFMRXdabVpXUnRjMmN4RHpBTkJnTlZCQU1UQm1abFpHMXpaekVQTUEwR0ExVUVLUk1H\nClptVmtiWE5uTVNZd0pBWUpLb1pJaHZjTkFRa0JGaGRoWkcxcGJrQm1aV1J2Y21Gd2NtOXFaV04w\nTG05eVo0SUoKQU9OUUhrZFBGeDVGTUJNR0ExVWRKUVFNTUFvR0NDc0dBUVVGQndNQ01Bc0dBMVVk\nRHdRRUF3SUhnREFOQmdrcQpoa2lHOXcwQkFRVUZBQU9CZ1FCU2hrTnZ5ZzdvclZxbVVXdHdmZHpW\nUS92QnFXWkp1bys2a0RwckJIYUJlVWdmClA2RExEUEVPN3JHZXlxaFVtb0U3dm!
 RVRHBGeE5
 mbC9E\nemtBOHJBRWVsMjdGUXZ5RVFkUDJFaHpPUjZlTldwT0cKdEVJU0RML1piRFgwZzRrSWR5YjVoWGZY\nbUJzbzQ1RmdPYkNGUkdsTlg5SUU1Ly95bmxCZU5CMzEvYlk1bmc9PQotLS0tLUVORCBDRVJUSUZJ\nQ0FURS0tLS0tCg==\n', u'i': 1, u'timestamp': 1426337840, u'msg_id': u'2015-924858d6-fe96-4939-b15c-e3a029235b43', u'crypto': u'x509', u'topic': u'org.fedoraproject.prod.buildsys.task.state.change', u'signature': u'fDoQERRuiAjpIlWCSYMxGtcnXzOI1HkNEyTZTVROcyf5upz2HdCLzSsoqQNZMcANV8iMTmKD1dva\nTofPHKdg6vRs9j9AVCplcz1KJQY0cnWrWUB1ObmFbLl+x+/FaxtVMw/WUFuA1hMyc/9BCom86zpg\np9yrNbtCqoZVY7wB4nY=\n', u'msg': {u'info': {u'children': [{u'parent': 9233187, u'channel_id': 12, u'start_time': u'2015-03-14 12:48:22.037546', u'state': 2, u'awaited': False, u'label': u'x86_64', u'priority': 19, u'completion_time': u'2015-03-14 12:57:11.982569', u'waiting': None, u'create_time': u'2015-03-14 12:48:10.693851', u'id': 9233192, u'owner': 2745, u'host_id': 160, u'arch': u'x86_64', u'method': u'createImage'}], u'parent': None, u'channel_i!
 d': 12, u
 'start_time': u'2015-03-14 12:48:09.903671', u'request': [u'Fedora-Docker-Base', u'22', [u'x86_64'], u'f22-candidate', u'http://kojipkgs.fedoraproject.org/mash/branched-20150314/22/$arch/os/', {u'kickstart': u'work/cli-image/1426337265.628317.mihniGVK/fedora-docker-base-4276571.ks', u'disk_size': u'3', u'format': [u'docker'], u'scratch': True, u'repo': [u'http://kojipkgs.fedoraproject.org/mash/branched-20150314/22/$arch/os/'], u'release': u'20150314', u'distro': u'Fedora-20'}], u'state': 2, u'awaited': None, u'method': u'image', u'priority': 20, u'completion_time': u'2015-03-14 12:57:20.589813', u'waiting': False, u'create_time': u'2015-03-14 12:47:45.835651', u'result': u'Scratch image build results in: /mnt/koji/work/tasks/3192/9233192', u'owner': 2745, u'host_id': 160, u'label': None, u'arch': u'noarch', u'id': 9233187}, u'old': u'OPEN', u'attribute': u'state', u'method': u'image', u'instance': u'primary', u'owner': u'masher', u'new': u'CLOSED', u'srpm': u'Fedora-Docker-!
 Base', u'
 id': 9233187}}, 'topic': u'org.fedoraproject.prod.buildsys.task.state.change'}

Process Details
---------------
host:     fedimg01.phx2.fedoraproject.org
PID:      6489
name:     fedmsg-hub
command:  /usr/bin/python /usr/bin/fedmsg-hub

Callstack that lead to the logging statement
--------------------------------------------
  File "/usr/lib64/python2.7/threading.py", line 784 in __bootstrap
    self.__bootstrap_inner()
  File "/usr/lib64/python2.7/threading.py", line 811 in __bootstrap_inner
    self.run()
  File "/usr/lib64/python2.7/threading.py", line 764 in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib64/python2.7/site-packages/twisted/python/threadpool.py", line 167 in _worker
    result = context.call(ctx, function, *args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/twisted/python/context.py", line 118 in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/usr/lib64/python2.7/site-packages/twisted/python/context.py", line 81 in callWithContext
    return func(*args,**kw)
  File "/usr/lib/python2.7/site-packages/moksha/hub/api/consumer.py", line 192 in _work
    self.log.exception(message)
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/moksha/hub/api/consumer.py", line 190, in _work
    self.consume(message)
  File "/usr/lib/python2.7/site-packages/fedimg/consumers.py", line 106, in consume
    fedimg.uploader.upload(self._get_upload_urls(builds))
  File "/usr/lib/python2.7/site-packages/fedimg/consumers.py", line 60, in _get_upload_urls
    upload_files.append(get_rawxz_url(task_result))
  File "/usr/lib/python2.7/site-packages/fedimg/util.py", line 52, in get_rawxz_url
    file_name = [f for f in task_result['files'] if f.endswith('.raw.xz')][0]
IndexError: list index out of range
```

**Comment by ralphbean** [comment]
Further information:

- Triggered by [this message](https://apps.fedoraproject.org/datagrepper/id?id=2015-924858d6-fe96-4939-b15c-e3a029235b43&size=extra-large)
- Which is [this build](http://koji.fedoraproject.org/koji/taskinfo?taskID=9233187)
- Which [doesn't have any .raw.xz files](http://koji.fedoraproject.org/koji/taskinfo?taskID=9233192)

... which should be somehow detected and ignored/handled instead of bubbling up to an error email.

**Comment by oddshocks** [comment]
Improvements/fixes to the createImage task filtering code in the recent release should have fixed this, but looking at the error logs, that's not the case. I'll have to poke around a bit more. Not sure why the filtering code isn't working properly. Maybe I made a mistake in my conditional logic (though I thought I had fixed that).

**Comment by oddshocks** [comment]
Oops -- I accidentally opened and later closed #23, which was a duplicate of this. My bad!
